Library/src/Multiplayer/EventBus.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "CSP/Multiplayer/EventBus.h"
+
+#include "CSP/Systems/SystemBase.h"
+#include "CallHelpers.h"
+#include "Debug/Logging.h"
+#include "Memory/Memory.h"
+#include "Multiplayer/SignalR/SignalRConnection.h"
+#include "NetworkEventManagerImpl.h"
+
+namespace csp::multiplayer
+{
+
+extern ErrorCode ParseError(std::exception_ptr Exception);
+
+constexpr const uint64_t ALL_CLIENTS_ID = -1;
+
+EventBus::~EventBus()
+{
+	CSP_DELETE(MultiplayerConnectionInst);
```

Comment:
I dont think we should be deleting this here

Reply:
done